config_management/domains/views.py

```diff
@@ -0,0 +1,69 @@
+from django.views.generic import View
+from .models import Domain
+from django.http.response import JsonResponse
+import json
+
+# Create your views here.
+class DomainView(View):
+
+    def get(self, request, d_id):
+
+        if d_id == 'all':
+            all_domains = Domain.objects.all()
+
+            domain_info = []
+            domain_names = []
+            domain_ids = []
+
+            for i in all_domains:
+                domain_info.append((i.get_info()['id'], i.get_info()['name']))
+                domain_names.append(i.get_info()['name'])
+                domain_ids.append(i.get_info()['id'])
+
+            args = {'domain_info':domain_info, 'domain_names':domain_names, 'domain_ids':domain_ids}
+
+            return JsonResponse(args)
+
+        else:
+
+            try:
+                domain_object = Domain.objects.get(id=d_id)
+            except:
+                return JsonResponse({'error_message':'The requested domain does not exist'})
+                
+            domain_name = domain_object.get_info()['name']
```

Comment:
Why are we gathering and reformatting attributes for response when it's already being done in get_info?